src/sql/workbench/contrib/notebook/browser/calloutDialog/linkCalloutDialog.ts

```diff
@@ -121,18 +129,41 @@ export class LinkCalloutDialog extends CalloutDialog<ILinkCalloutDialogOptions>
 		this._register(styler.attachInputBoxStyler(this._linkUrlInputBox, this._themeService));
 	}
 
+	protected onAccept(e?: StandardKeyboardEvent) {
+		e.stopPropagation();
+		// Without calling preventDefault, text cell will insert an extra newline when pressing enter on dialog
+		e.preventDefault();
+		this.insert();
+	}
+
+	protected onClose(e?: StandardKeyboardEvent) {
+		e.stopPropagation();
+		e.preventDefault();
+		this.cancel();
+	}
 	public insert(): void {
 		this.hide();
+		let label = strings.escape(this._linkTextInputBox.value);
```

Comment:
I don't understand this comment. Can you explain?

Reply:
If you have a label such as `My_]_label` then this will break the markdown (it'll be something like `[My_]_label](...)`. Similarly with ) in the URL portion.

Right now strings.escape is just escaping a few specific characters - not those ones